### ISSUE TITLE
Appium 1.3.1 "Error: App did not have elements" if an Alert show up with an iOS application

### ISSUE THREAD

**Issue by LimeDang** [appium/appium#4032]
Appium 1.3.1
Xcode 6.1
iOS 8.1

When I start Appium I see my iOS application started successfully and display an alert to ask or confirm somethings, right after that Appium will get an error "Error: App did not have elements" and exit.

I have tried dismiss the alert myself and Appium work, no more "Error: App did not have elements".
So I think this is a bug from Appium, if Appium does not have this error and start successfully then I can execute an UIAutomation script to dismiss this alert.

Here is the full log:

[36minfo[39m: Welcome to Appium v1.3.1 (REV 1160ce02bb89c354cb99317985123acf39f0e7d3)
[36minfo[39m: Appium REST http interface listener started on 0.0.0.0:4723
[36minfo[39m: [debug] Non-default server args: {"udid":"20846aec46cd64589f6a2b356ef2390b2bc6bc6c","fullReset":true,"defaultCommandTimeout":3600,"tmpDir":"/Users/admin/Appium_Temp"}
[36minfo[39m: Console LogLevel: debug
[36minfo[39m: [37m-->[39m [37mGET[39m [37m/wd/hub/status[39m [90m{}[39m
[36minfo[39m: [debug] Responding to client with success: {"status":0,"value":{"build":{"version":"1.3.1","revision":"1160ce02bb89c354cb99317985123acf39f0e7d3"}}}
[36minfo[39m: [37m<-- GET /wd/hub/status [39m[32m200[39m[90m 6.514 ms - 104[39m [90m{"status":0,"value":{"build":{"version":"1.3.1","revision":"1160ce02bb89c354cb99317985123acf39f0e7d3"}}}[39m
[36minfo[39m: [37m-->[39m [37mPOST[39m [37m/wd/hub/session[39m [90m{"desiredCapabilities":{"app":"/liemdang/Applications/eclipse/WP/branch_2.0.0.34/MobileLibs/target/classes/drivers/ios/ATCShowRoom.app","platformName":"iOS","deviceName":"iPhone 5s.mobile testing (8.1) [20846aec46cd64589f6a2b356ef2390b2bc6bc6c]"}}[39m
[36minfo[39m: Client User-Agent string: Apache-HttpClient/4.3.3 (java 1.5)
[36minfo[39m: [debug] Using local app from desired caps: /liemdang/Applications/eclipse/WP/branch_2.0.0.34/MobileLibs/target/classes/drivers/ios/ATCShowRoom.app
[36minfo[39m: [debug] Creating new appium session e75a9060-243a-404d-91d1-999dec79396d
[36minfo[39m: [debug] Removing any remaining instruments sockets
[36minfo[39m: [debug] Cleaned up instruments socket /Users/admin/Appium_Temp/instruments_sock
[36minfo[39m: [debug] Setting Xcode folder
[36minfo[39m: [debug] Setting Xcode version
[36minfo[39m: [debug] Setting iOS SDK Version
[36minfo[39m: [debug] Getting sdk version from xcrun with a timeout
[36minfo[39m: [debug] iOS SDK Version set to 8.1
[36minfo[39m: [debug] Not checking whether simulator is available since we're on a real device
[36minfo[39m: [debug] Detecting automation tracetemplate
[36minfo[39m: [debug] Not auto-detecting udid, running on sim
[36minfo[39m: [debug] Could not parse plist file (as binary) at /liemdang/Applications/eclipse/WP/branch_2.0.0.34/MobileLibs/target/classes/drivers/ios/ATCShowRoom.app/en.lproj/Localizable.strings
[36minfo[39m: Will try to parse the plist file as XML
[36minfo[39m: [debug] Could not parse plist file (as XML) at /liemdang/Applications/eclipse/WP/branch_2.0.0.34/MobileLibs/target/classes/drivers/ios/ATCShowRoom.app/en.lproj/Localizable.strings
[33mwarn[39m: Could not parse app Localizable.strings; assuming it doesn't exist
[36minfo[39m: [debug] Getting bundle ID from app
[36minfo[39m: [debug] Parsed app Info.plist (as binary)
[36minfo[39m: [debug] Creating instruments
[36minfo[39m: On some xcode 6 platforms, instruments-without-delay does not work. If you experience this, you will need to re-run appium with the --native-instruments-lib flag
[36minfo[39m: [debug] Preparing uiauto bootstrap
[36minfo[39m: [debug] Dynamic bootstrap dir: /Users/admin/Library/Application Support/appium/bootstrap
[36minfo[39m: [debug] Dynamic env: {"nodePath":"/usr/local/Cellar/node/0.10.32/bin/node","commandProxyClientPath":"/usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js","instrumentsSock":"/Users/admin/Appium_Temp/instruments_sock","interKeyDelay":null,"justLoopInfinitely":false,"autoAcceptAlerts":false}
[36minfo[39m: [debug] Dynamic bootstrap code: // This file is automatically generated. Do not manually modify!
...
[36minfo[39m: [debug] Dynamic bootstrap path: /Users/admin/Library/Application Support/appium/bootstrap/bootstrap-1068c1a0452ee570.js
[36minfo[39m: [debug] Reusing dynamic bootstrap: /Users/admin/Library/Application Support/appium/bootstrap/bootstrap-1068c1a0452ee570.js
[36minfo[39m: [debug] Getting device string from opts: {"forceIphone":false,"forceIpad":false,"xcodeVersion":"6.1","iOSSDKVersion":"8.1","deviceName":"iPhone 5s.mobile testing (8.1) [20846aec46cd64589f6a2b356ef2390b2bc6bc6c]","platformVersion":null}
[36minfo[39m: [debug] fixDevice is on
[36minfo[39m: [debug] Final device string is: 'iPhone 5s.mobile testing (8.1) [20846aec46cd64589f6a2b356ef2390b2bc6bc6c] (8.1 Simulator)'
[36minfo[39m: [debug] Not setting device type since we're on a real device
[36minfo[39m: [debug] Checking whether we need to set app preferences
[36minfo[39m: [debug] Not setting iOS and app preferences since we're on a real device
[36minfo[39m: [debug] Running ios sim reset flow
[36minfo[39m: [debug] Killing the simulator process
[36minfo[39m: [debug] Killall iOS Simulator
[36minfo[39m: [debug] Killing any other simulator daemons
[36minfo[39m: [debug] On a real device; cannot clean device state
[36minfo[39m: [debug] Not setting locale because we're using a real device
[36minfo[39m: [debug] No iOS / app preferences to set
[36minfo[39m: [debug] Starting iOS device log capture via deviceconsole
[36minfo[39m: [debug] Not pre-launching simulator
[36minfo[39m: [debug] Creating iDevice object with udid 20846aec46cd64589f6a2b356ef2390b2bc6bc6c
[36minfo[39m: [debug] Couldn't find ideviceinstaller, trying built-in at /usr/local/lib/node_modules/appium/build/libimobiledevice-macosx/ideviceinstaller
[36minfo[39m: [debug] Checking app install status using: /usr/local/lib/node_modules/appium/build/fruitstrap/fruitstrap isInstalled --id 20846aec46cd64589f6a2b356ef2390b2bc6bc6c --bundle com.kms.showroom.ipad
[36minfo[39m: [debug] App is not installed. Will try to install the app.
[36minfo[39m: [debug] Installing app using cmd: /usr/local/lib/node_modules/appium/build/fruitstrap/fruitstrap install --id 20846aec46cd64589f6a2b356ef2390b2bc6bc6c --bundle "/liemdang/Applications/eclipse/WP/branch_2.0.0.34/MobileLibs/target/classes/drivers/ios/ATCShowRoom.app"
[36minfo[39m: [debug] Starting command proxy.
[36minfo[39m: [debug] Instruments socket server started at /Users/admin/Appium_Temp/instruments_sock
[36minfo[39m: [debug] Starting instruments
[36minfo[39m: [debug] Instruments is at: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments
[36minfo[39m: Launching instruments
[36minfo[39m: [debug] Attempting to run app on real device with UDID 20846aec46cd64589f6a2b356ef2390b2bc6bc6c
[36minfo[39m: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /Users/admin/Appium_Temp/appium-instruments/instrumentscli0.trace -w 20846aec46cd64589f6a2b356ef2390b2bc6bc6c com.kms.showroom.ipad -e UIASCRIPT "/Users/admin/Library/Application Support/appium/bootstrap/bootstrap-1068c1a0452ee570.js" -e UIARESULTSPATH /Users/admin/Appium_Temp/appium-instruments
[36minfo[39m: [debug] And extra without-delay env: {}
[36minfo[39m: [debug] And launch timeouts (in ms): {"global":90000}
[36minfo[39m: [debug] [33m[INST STDERR] 2014-11-12 10:17:38.595 instruments[829:15578] WebKit Threading Violation - initial use of WebKit from a secondary thread.[39m
[36minfo[39m: Instruments is ready to receive commands
[36minfo[39m: [debug] Instruments launched. Starting poll loop for new commands.
[36minfo[39m: [debug] Setting bootstrap config keys/values
[36minfo[39m: [debug] Pushing command to appium work queue: "target = $.target();\nau = $;\n$.isVerbose = true;\n"
[36minfo[39m: [debug] Socket data received (2 bytes)
[36minfo[39m: [debug] Socket data being routed.
[36minfo[39m: [debug] Sending command to instruments: target = $.target();
au = $;
$.isVerbose = true;

[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:40 +0000 Debug: evaluation finished[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:40 +0000 Debug: responding with:[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:40 +0000 Debug: Running system command #1: /usr/local/Cellar/node/0.10.32/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /Users/admin/Appium_Temp/instruments_sock 2,{"status":0,"value":true}...[39m
[36minfo[39m: [debug] Socket data received (27 bytes)
[36minfo[39m: [debug] Socket data being routed.
[36minfo[39m: [debug] Got result from instruments: {"status":0,"value":true}
[36minfo[39m: [debug] Setting initial orientation to PORTRAIT
[36minfo[39m: [debug] Pushing command to appium work queue: "au.setScreenOrientation('PORTRAIT')"
[36minfo[39m: [debug] Sending command to instruments: au.setScreenOrientation('PORTRAIT')
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:41 +0000 Debug: Got new command 1 from instruments: au.setScreenOrientation('PORTRAIT')[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:41 +0000 Debug: evaluating au.setScreenOrientation('PORTRAIT')[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:41 +0000 Debug: target.setDeviceOrientation("1")[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:42 +0000 Debug: evaluation finished[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:42 +0000 Debug: responding with:[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:42 +0000 Debug: Running system command #2: /usr/local/Cellar/node/0.10.32/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /Users/admin/Appium_Temp/instruments_sock 2,{"status":0,"value":"PORTRAIT"}...[39m
[36minfo[39m: [debug] Socket data received (33 bytes)
[36minfo[39m: [debug] Socket data being routed.
[36minfo[39m: [debug] Got result from instruments: {"status":0,"value":"PORTRAIT"}
[36minfo[39m: [debug] Waiting for app source to contain elements
[36minfo[39m: [debug] Pushing command to appium work queue: "au.mainApp().getTreeForXML()"
[36minfo[39m: [debug] Sending command to instruments: au.mainApp().getTreeForXML()
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:43 +0000 Debug: Got new command 2 from instruments: au.mainApp().getTreeForXML()[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:43 +0000 Debug: evaluating au.mainApp().getTreeForXML()[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:46 +0000 Debug: evaluation finished[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:46 +0000 Debug: responding with:[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:46 +0000 Debug: Running system command #3: /usr/local/Cellar/node/0.10.32/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /Users/admin/Appium_Temp/instruments_sock 2,{"status":0,"value":{"UIAApplication":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":false,"valid":false,"vi...[39m
[36minfo[39m: [debug] Socket data received (271 bytes)
[36minfo[39m: [debug] Socket data being routed.
[36minfo[39m: [debug] Got result from instruments: {"status":0,"value":{"UIAApplication":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":false,"valid":false,"visible":false,"hint":null,"path":"/0","x":2.5696237245e-314,"y":2.5686474587e-314,"width":1.061256928e-314,"height":3.3468680043e-314},">":[]}}}
[36minfo[39m: [debug] Waited for 3959ms so far
[36minfo[39m: [debug] Pushing command to appium work queue: "au.mainApp().getTreeForXML()"
[36minfo[39m: [debug] Sending command to instruments: au.mainApp().getTreeForXML()
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:47 +0000 Debug: Got new command 3 from instruments: au.mainApp().getTreeForXML()[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:47 +0000 Debug: evaluating au.mainApp().getTreeForXML()[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:50 +0000 Debug: evaluation finished[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:50 +0000 Debug: responding with:[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:50 +0000 Debug: Running system command #4: /usr/local/Cellar/node/0.10.32/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /Users/admin/Appium_Temp/instruments_sock 2,{"status":0,"value":{"UIAApplication":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":false,"valid":false,"vi...[39m
[36minfo[39m: [debug] Socket data received (271 bytes)
[36minfo[39m: [debug] Socket data being routed.
[36minfo[39m: [debug] Got result from instruments: {"status":0,"value":{"UIAApplication":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":false,"valid":false,"visible":false,"hint":null,"path":"/0","x":2.5696237245e-314,"y":2.5686386446e-314,"width":1.061256928e-314,"height":3.3468680043e-314},">":[]}}}
[36minfo[39m: [debug] Waited for 8042ms so far
[36minfo[39m: [debug] Pushing command to appium work queue: "au.mainApp().getTreeForXML()"
[36minfo[39m: [debug] Sending command to instruments: au.mainApp().getTreeForXML()
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:51 +0000 Debug: Got new command 4 from instruments: au.mainApp().getTreeForXML()[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:51 +0000 Debug: evaluating au.mainApp().getTreeForXML()[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:54 +0000 Debug: evaluation finished[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:54 +0000 Debug: responding with:[39m
[36minfo[39m: [debug] [32m[INST] 2014-11-12 03:17:54 +0000 Debug: Running system command #5: /usr/local/Cellar/node/0.10.32/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /Users/admin/Appium_Temp/instruments_sock 2,{"status":0,"value":{"UIAApplication":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":false,"valid":false,"vi...[39m
[36minfo[39m: [debug] Socket data received (271 bytes)
[36minfo[39m: [debug] Socket data being routed.
[36minfo[39m: [debug] Got result from instruments: {"status":0,"value":{"UIAApplication":{"@":{"name":null,"label":null,"value":null,"dom":null,"enabled":false,"valid":false,"visible":false,"hint":null,"path":"/0","x":2.5696237245e-314,"y":2.5681063897e-314,"width":1.061256928e-314,"height":3.3468680043e-314},">":[]}}}
[36minfo[39m: [debug] Condition unmet after 12137ms. Timing out.
[36minfo[39m: [debug] Cleaning up appium session
[36minfo[39m: [debug] Error: App did not have elements
    at /usr/local/lib/node_modules/appium/lib/devices/ios/ios.js:446:28
    at Object.cb [as callback] (/usr/local/lib/node_modules/appium/lib/devices/ios/ios-controller.js:247:7)
    at next (/usr/local/lib/node_modules/appium/node_modules/async/lib/async.js:801:43)
    at /usr/local/lib/node_modules/appium/node_modules/async/lib/async.js:32:16
    at exports.respond (/usr/local/lib/node_modules/appium/lib/devices/common.js:28:9)
    at null.<anonymous> (/usr/local/lib/node_modules/appium/lib/devices/ios/ios.js:1434:18)
    at getResultAndSendNext (/usr/local/lib/node_modules/appium/node_modules/appium-uiauto/lib/command-proxy.js:146:20)
    at Socket.<anonymous> (/usr/local/lib/node_modules/appium/node_modules/appium-uiauto/lib/command-proxy.js:88:7)
    at Socket.emit (events.js:117:20)
    at _stream_readable.js:943:16
[36minfo[39m: [debug] Responding to client with error: {"status":33,"value":{"message":"A new session could not be created. (Original error: App did not have elements)","origValue":"App did not have elements"},"sessionId":null}
[36minfo[39m: [37m<-- POST /wd/hub/session [39m[31m500[39m[90m 39007.711 ms - 172[39m [90m[39m

**Comment by hallbeck** [comment]
Are you saying that if we set a capability waitForAppScript to true it will work?

Is that the exact capability to use? I ask so I can start running my automation again. We have been stuck running the tests manually for ages.
and what build should we use?

**Comment by LimeDang** [comment]
@jlipps @sebv if we added waitForAppScript: true, message "App did not have elements" will no longer appear, but the instruments still can not get any objects from the app, the app or the instruments is freeze, I can not do any actions manully on the app.
I think that the issue may come from the Xcode Instruments, it only happen on iOS8 and Xcode6.
I did some research and workarounds are: turn off wifi of the device or disable all NSLog statements in source code of the ios app, and the Xcode Instruments works as normal (no more freeze).

Do you have any ideas or other solutions for this issue? I am using latest version 1.3.4

**Comment by jlipps** [comment]
merging this into #4178

**Comment by dianamenezes** [comment]
I recently upgraded from 1.3.7 to 1.4.1 and am seeing this issue for iOS simulator: 
error: Failed to start an Appium session, err was: Error: App did not have elements

I added waitForAppActivity: true, and it works fine. I want to clarify this is expected?

**Comment by LiudasTumasonis** [comment]
@dianamenezes  - where exactly did you set the waitForAppActivity: true?

I am having the exact same issue right now and it used to work just fine, but now it won't even start the test because the first thing that happens when the app opens is a an alert and unless i manually accept it - it won't move any further... Thanks for your help!

**Comment by dianamenezes** [comment]
@LiudasTumasonis 
I set it in the iOS capabilities section

something like this:
desired_caps = {
          caps:       {
              platformName:  'iOS',
              platformVersion: @version,
              deviceName:    'iPhone Simulator',
              sendKeyStrategy: 'grouped',
              waitForAppScript: true,
              app:           app_path
          }
      }

**Comment by LiudasTumasonis** [comment]
@dianamenezes  - thank you so much!
I'd get you flowers if I could... I've spent past 4 hours trying all kinds of things to fix this and this waitForAppScript = true did the trick!

Thanks again

**Comment by dianamenezes** [comment]
@LiudasTumasonis 
Thank you. 
I am glad you got it working.

**Comment by Sofrus** [comment]
@LiudasTumasonis 
I would like to know there is the iOS capabilities section? I start appium server from appium.app in macos. In which file i must add waitForAppScript: true?
Thank you an advance.

**Comment by alangithubtrader** [comment]
@LiudasTumasonis 

Awesome. This worked for me too. Thank you so much for helping us out!

**Comment by firebear** [comment]
With @dianamenezes 's solution , I succeeded to skip the error infomation of 'App did not have elements'. However I got another issue which I am not sure if has nothing to do with the waitForAppScript option.
Any help would be appreciated.

~~~
PHPUnit 4.7.0 by Sebastian Bergmann and contributors.

E

Time: 59.31 seconds, Memory: 32.25Mb

There was 1 error:

1) IOSTest::testSanity
BadMethodCallException: The command http://172.16.2.46:4723/wd/hub/session is not recognized by the server.

/var/www/devbox/Emma/vendor/appium/php-client/PHPUnit/Extensions/AppiumTestCase/Driver.php:39
/var/www/devbox/Emma/vendor/appium/php-client/PHPUnit/Extensions/AppiumTestCase/SessionStrategy/Isolated.php:30

FAILURES!
Tests: 1, Assertions: 0, Errors: 1.
~~~

My screen capture of appium.app

~~~

info: [debug] [INST] Waiting for device to boot...

info: [debug] [INST STDERR] 2015-08-03 12:30:38.986 instruments[26437:155127] WebKit Threading Violation - initial use of WebKit from a secondary thread.

info: Executing: xcrun simctl erase C21F590A-825E-427D-BD38-16C85ED1AC6C with timeout 2000

info: Executing: xcrun simctl erase C21F590A-825E-427D-BD38-16C85ED1AC6C with timeout 2000

info: [debug] Cleaning app data files

warn: Applications directory /Users/firebear/Library/Developer/CoreSimulator/Devices/C21F590A-825E-427D-BD38-16C85ED1AC6C/data/Applications doesn't exist. Have you run this simulator before?
info: Couldn't find app directories to delete. Probably it's not installed
info: [debug] Cleaning up appium session

info: --> GET /wd/hub/status {}

info: [debug] Responding to client with success: {"status":0,"value":{"build":{"version":"1.4.8","revision":"c8179bd8352d3c62d98fa58b324491230df9342a"}}}

info: <-- GET /wd/hub/status 200 2.158 ms - 104 {"status":0,"value":{"build":{"version":"1.4.8","revision":"c8179bd8352d3c62d98fa58b324491230df9342a"}}}

info: Instruments is ready to receive commands

info: [debug] Instruments launched. Starting poll loop for new commands.
info: [debug] Setting bootstrap config keys/values
info: [debug] Pushing command to appium work queue: "target = $.target();\nau = $;\n$.isVerbose = true;\n"

info: [debug] Socket data received (2 bytes)
info: [debug] Socket data being routed.
info: [debug] Sending command to instruments: target = $.target();
au = $;
$.isVerbose = true;


info: [debug] [INST] 2015-08-03 04:31:14 +0000 Debug: evaluation finished

info: [debug] [INST] 2015-08-03 04:31:14 +0000 Debug: responding with:
info: [debug] [INST] 2015-08-03 04:31:14 +0000 Debug: Running system command #1: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /tmp/instruments_sock 2,{"status":0,"value":true}...

info: [debug] Socket data received (27 bytes)

info: [debug] Socket data being routed.
info: [debug] Got result from instruments: {"status":0,"value":true}
info: [debug] Setting initial orientation to PORTRAIT
info: [debug] Pushing command to appium work queue: "au.setScreenOrientation('PORTRAIT')"
info: [debug] Sending command to instruments: au.setScreenOrientation('PORTRAIT')

info: [debug] [INST] 2015-08-03 04:31:15 +0000 Debug: Got new command 1 from instruments: au.setScreenOrientation('PORTRAIT')

info: [debug] [INST] 2015-08-03 04:31:15 +0000 Debug: evaluating au.setScreenOrientation('PORTRAIT')

info: [debug] [INST] 2015-08-03 04:31:15 +0000 Debug: target.setDeviceOrientation("1")

info: [debug] Socket data received (33 bytes)

info: [debug] Socket data being routed.
info: [debug] Got result from instruments: {"status":0,"value":"PORTRAIT"}
info: [debug] Using custom script to wait for app start:true
info: [debug] Pushing command to appium work queue: "try{\ntrue\n} catch(err) { $.log(\"waitForAppScript err: \" + error); false; };"
info: [debug] Sending command to instruments: try{
true
} catch(err) { $.log("waitForAppScript err: " + error); false; };

info: [debug] [INST] 2015-08-03 04:31:17 +0000 Debug: evaluation finished

info: [debug] [INST] 2015-08-03 04:31:17 +0000 Debug: responding with:

info: [debug] Socket data received (27 bytes)

info: [debug] Socket data being routed.
info: [debug] Got result from instruments: {"status":0,"value":true}
info: [debug] Device launched! Ready for commands

info: [debug] Setting command timeout to the default of 7200 secs
info: [debug] Appium session started with sessionId null
info: <-- POST /wd/hub/session 303 59155.948 ms - 42 
info: --> GET /wd/hub/session/null {}
info: <-- GET /wd/hub/session/null 404 0.778 ms - 40 

info: [debug] [INST] 2015-08-03 04:31:17 +0000 Debug: Running system command #2: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /tmp/instruments_sock 2,{"status":0,"value":"PORTRAIT"}...

info: [debug] [INST] 2015-08-03 04:31:18 +0000 Debug: Got new command 2 from instruments: try{
       true
} catch(err) { $.log("waitForAppScript err: " + error); false; };

info: [debug] [INST] 2015-08-03 04:31:18 +0000 Debug: evaluating try{
       true
} catch(err) { $.log("waitForAppScript err: " + error); false; };

info: [debug] [INST] 2015-08-03 04:31:18 +0000 Debug: evaluation finished

info: [debug] [INST] 2015-08-03 04:31:18 +0000 Debug: responding with:

info: [debug] [INST] 2015-08-03 04:31:18 +0000 Debug: Running system command #3: /Applications/Appium.app/Contents/Resources/node/bin/node /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /tmp/instruments_sock 2,{"status":0,"value":true}...

info: --> GET /wd/hub/status {}

info: [debug] Responding to client with success: {"status":0,"value":{"build":{"version":"1.4.8","revision":"c8179bd8352d3c62d98fa58b324491230df9342a"}}}
info: <-- GET /wd/hub/status 200 0.743 ms - 104 {"status":0,"value":{"build":{"version":"1.4.8","revision":"c8179bd8352d3c62d98fa58b324491230df9342a"}}}

info: --> GET /wd/hub/status {}

~~~

**Comment by cjtcwyk** [comment]
ocurring the same issue as LimeDang, the autoAcceptAlerts is not working. 
My env XC 6.1, Appium 1.4.7/1.4, java-client 2.2.0. When I added capabilities.setCapability("autoAcceptAlerts", true); to the capabilities, message "App did not have elements" will no longer appear, but the instruments still can not get any objects from the app and response info: [debug] Responding to client with error: {"status":7,"value":{"message":"An element could not be located on the page using the given search parameters."}, the app freeze at the welcome Image Windows. Now I can't do any test with Appium.

**Comment by dianamenezes** [comment]
@cjtcwyk 
Try this, remove the autoAcceptAlert cap and add waitForAppScript with a delay. You can increase/decrease the delay as needed. We just tried and seems to work for us where autoAcceptAlerts does not. 

desired_caps = {
caps: {
platformName: 'iOS',
platformVersion: @version,
deviceName: device_sim,
sendKeyStrategy: 'grouped',
app: app_path
waitForAppScript: '$.delay(5000); $.acceptAlert();'
}
}

**Comment by cjtcwyk** [comment]
@dianamenezes thanks for the suggestion, but it's not works to me : (
looks like I should do some delay in my AUT(https://github.com/appium/appium/issues/4178) when the app luanch.

**Comment by dianamenezes** [comment]
@cjtcwyk  
waitForAppScript: '$.delay(5000); $.acceptAlert();' has the delay. You can increase the delay time as required for your AUT?

**Comment by jlipps** [comment]
@dianamenezes this delay is not sufficient; sometimes the app itself needs to delay the permissions check in order for instruments to not be blocked by this race condition. hopefully that's what @cjtcwyk is attempting to do now.

**Comment by cjtcwyk** [comment]
@jlipps that's what i mean, but what make me confuse is the AUT has NO pop windows or any system permissions check when the app launch, it just freeze at WelcomeView which is the first View pop up on the screen when the app launch, and I don't think the WelcomeView would blocked launch. Any suggestion would be helpful.

**Comment by ShankarGargXebia** [comment]
waitForAppScript worked for me as well. Thanks!!

**Comment by ChiWhiteSox56** [comment]
Adding both desired capabilities mentioned in this thread worked for me: 

    caps.setCapability("autoAcceptAlerts", true);
    caps.setCapability("waitForAppScript", true);

Many thanks to everyone who contributed here!
